spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -144,60 +155,71 @@ def setup(self, mute_inline_plotting=None, show_plot_outline=None,
         self.setLayout(layout)
 
     def setup_toolbar(self):
-        """Setup the toolbar"""
-        savefig_btn = create_toolbutton(
-                self, icon=ima.icon('filesave'),
-                tip=_("Save Image As..."),
-                triggered=self.save_figure)
+        """Setup the toolbar."""
+        self.savefig_btn = create_toolbutton(
+            self, icon=ima.icon('filesave'),
+            tip=_("Save plot as..."),
+            triggered=self.save_figure)
+        self.savefig_btn.shortcut_data = ('plots', 'save')
+        self.savefig_btn.installEventFilter(self)
 
         saveall_btn = create_toolbutton(
-                self, icon=ima.icon('save_all'),
-                tip=_("Save All Images..."),
-                triggered=self.save_all_figures)
+            self, icon=ima.icon('save_all'),
+            tip=_("Save all plots..."),
+            triggered=self.save_all_figures)
+        saveall_btn.shortcut_data = ('plots', 'save all')
+        saveall_btn.installEventFilter(self)
 
         copyfig_btn = create_toolbutton(
             self, icon=ima.icon('editcopy'),
-            tip=(_("Copy plot to clipboard as image (%s)") %
-                 CONF.get_shortcut('plots', 'copy')),
+            tip=_("Copy plot to clipboard as image"),
             triggered=self.copy_figure)
+        copyfig_btn.shortcut_data = ('plots', 'copy')
+        copyfig_btn.installEventFilter(self)
 
-        closefig_btn = create_toolbutton(
-                self, icon=ima.icon('editclear'),
-                tip=_("Remove image"),
-                triggered=self.close_figure)
+        self.closefig_btn = create_toolbutton(
```

Comment:
Not work for this PR, but should not we provide this information to the create tool button func?

So that we do not have to do this 

`self.closefig_btn.shortcut_data = ('plots', 'close')`  ?

Thoughts @jnsebgosselin if you think this would be useful?

Reply:
I don't know... That is the trick I used to update the tooltips of the toolbuttons for the plots plugin, but I think we need to think of how we could achieve the same results, but for all plugins that have toolbuttons with customizable shortcuts.